plugins/outputs/bigquery/bigquery_test.go

```diff
@@ -0,0 +1,96 @@
+package bigquery
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/http/httputil"
+	"os"
+	"strings"
+	"testing"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/api/option"
+)
+
+const (
+	successfulResponse = "{\"kind\": \"bigquery#tableDataInsertAllResponse\"}"
+)
+
+var testingHost string
+
+func TestMain(t *testing.M) {
+	srv := localBigQueryServer(t)
+
+	testingHost = strings.ReplaceAll(srv.URL, "http://", "")
+
+	defer srv.Close()
+
+	os.Exit(t.Run())
+}
+
+func TestConnect(t *testing.T) {
+	b := &BigQuery{
+		Project: "test-project",
+		Dataset: "test-dataset",
+	}
+
+	b.setUpTestClient()
+	err := b.Connect()
+	require.NoError(t, err)
+
+}
+
+func TestWrite(t *testing.T) {
+	b := &BigQuery{
+		Project: "test-project",
+		Dataset: "test-dataset",
+	}
+
+	mockMetrics := testutil.MockMetrics()
+	b.setUpTestClient()
+	b.Connect()
+	err := b.Write(mockMetrics)
+	require.NoError(t, err)
+}
+
```

Comment:
Could you add a test where you actually compare what is sent to the server with what you expect? Do you have an example of the on-wire format?